java/maven.grammar/src/org/netbeans/modules/maven/grammar/MavenProjectGrammar.java

```diff
@@ -464,6 +468,70 @@ protected Enumeration<GrammarResult> getDynamicValueCompletion(String path, Hint
                 return Collections.enumeration(elems);
             }
         }
+        // version property completion
+        if (path.startsWith("/project/properties/") && path.indexOf("/", 20) == -1) {
```

Comment:
this should make it hopefully fairly safe for inclusion. The code path is guarded by the xpath (basically), so it doesn't affect any other completion code.

Reply:
Nitpick: I would store the string to variable/constant, then would use `length()` or leave a comment for `20`. I would add `// NOI18N` if possible.